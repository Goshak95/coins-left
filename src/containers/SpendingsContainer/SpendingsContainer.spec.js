import React from 'react'
import { shallow } from 'enzyme'
import { SpendingsContainer } from '.'

describe('Spendings container', () => {
  const props = {
    spendings: {
      error: null,
      isLoading: false,
      spendingsList: [],
    },
    categories: { error: null, categoriesData: [] },
    icons: { error: null, iconsData: [] },
    getSpendings: () => {},
    addSpending: () => {},
    deleteSpending: () => {},
    editSpending: () => {},
    getCategories: () => {},
    addCategory: () => {},
    deleteCategory: () => {},
    getIcons: () => {},
  }

  describe('Spendings container initial render', () => {
    const mockFetchGetSpendings = jest.fn()
    const mockFetchGetCategories = jest.fn()
    const mockFetchGetIcons = jest.fn()

    const nextProps = {
      ...props,
      getSpendings: mockFetchGetSpendings,
      getCategories: mockFetchGetCategories,
      getIcons: mockFetchGetIcons,
    }

    const spendingsContainer = shallow(<SpendingsContainer {...nextProps} />)

    it('renders properly', () => {
      expect(spendingsContainer).toMatchSnapshot()
    })

    it("dispatches the 'getSpendings' method froms props", () => {
      expect(mockFetchGetSpendings).toHaveBeenCalledTimes(1)
    })

    it("dispatches the 'getCategories' method froms props", () => {
      expect(mockFetchGetCategories).toHaveBeenCalledTimes(1)
    })

    it("dispatches the 'getIcons' method froms props", () => {
      expect(mockFetchGetIcons).toHaveBeenCalledTimes(1)
    })
  })

  describe('Spendings container is loading', () => {
    const nextProps = { ...props, spendings: { ...props.spendings, isLoading: true } }
    const spendingsContainer = shallow(<SpendingsContainer {...nextProps} />)

    it('Renders properly', () => {
      expect(spendingsContainer).toMatchSnapshot()
    })

    it('Loader renders', () => {
      expect(spendingsContainer.find('Loader')).toHaveLength(1)
    })
  })

  describe('Spendings container Tabs render', () => {
    const nextProps = {
      ...props,
      spendings: {
        ...props.spendings,
        spendingsList: [
          { id: 1, name: 'some' },
          { id: 2, name: 'lists' },
        ],
      },
    }
    const spendingsContainer = shallow(<SpendingsContainer {...nextProps} />)

    it('Renders properly', () => {
      expect(spendingsContainer).toMatchSnapshot()
    })
  })

  describe('Spendings container on error', () => {
    const nextProps = {
      ...props,
      spendings: {
        ...props.spendings,
        error: 'Some error message',
      },
    }
    const spendingsContainer = shallow(<SpendingsContainer {...nextProps} />)

    it('Renders properly', () => {
      expect(spendingsContainer).toMatchSnapshot()
    })

    it('ErrorComponent renders', () => {
      expect(spendingsContainer.find('ErrorComponent')).toHaveLength(1)
    })
  })
})
