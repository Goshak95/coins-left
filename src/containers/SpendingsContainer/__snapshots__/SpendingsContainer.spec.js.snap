// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spendings container Spendings container Tabs render Renders properly 1`] = `
<Tabs>
  <Tab
    path="/spendings/today"
    title="Current spendings"
  >
    <DayCard
      actions={
        Object {
          "addCategory": [Function],
          "addSpending": [Function],
          "deleteCategory": [Function],
          "deleteSpending": [Function],
          "editSpending": [Function],
          "getCategories": [Function],
          "getIcons": [Function],
          "getSpendings": [Function],
        }
      }
      categories={Array []}
      icons={Array []}
      spendingsList={
        Array [
          Object {
            "id": 1,
            "name": "some",
          },
          Object {
            "id": 2,
            "name": "lists",
          },
        ]
      }
    />
  </Tab>
  <Tab
    path="/spendings/categories"
    title="Categories"
  >
    <h1>
      Categories!
    </h1>
  </Tab>
</Tabs>
`;

exports[`Spendings container Spendings container initial render renders properly 1`] = `""`;

exports[`Spendings container Spendings container is loading Renders properly 1`] = `
<Loader
  className="loader"
  color="#00BFFF"
  height={150}
  style={Object {}}
  timeout={0}
  type="ThreeDots"
  visible={true}
  width={150}
/>
`;

exports[`Spendings container Spendings container on error Renders properly 1`] = `
<ErrorComponent
  message="Some error message"
/>
`;
